### PR TITLE
Update permission checks in CacheCreateConfig and GetSemaphoreType tasks [HZ-2090]

### PR DESCRIPTION
**Pull request by kwart**
Permission type fixes. Follow up for #25509

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheCreateConfigMessageTask.java [hunk 1]
```diff
@@ -81,7 +83,8 @@ public String getServiceName() {
 
     @Override
     public Permission getRequiredPermission() {
-        return new ConfigPermission();
+        CacheConfig cacheConfig = parameters.cacheConfig.asCacheConfig(serializationService);
+        return new CachePermission(CacheUtil.getDistributedObjectName(cacheConfig.getName()), ActionConstants.ACTION_CREATE);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheCreateConfigMessageTask.java:87]: `ConfigPermission` seems a better fit since this message task adds the `CacheConfig` to the cluster members, it does not create the `Cache` proxy. What is the reason for this change?

Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheCreateConfigMessageTask.java:87]: Do I understand it correcctly, this action is always taken when a client wants to access a cache?
Then `CachePermission` makes more sense to me. We could change the action (supported are: `CREATE | DESTROY | PUT | REMOVE | READ | LISTEN`) or remove it completely (i.e. any granted CachePermission allows this task).
The `ConfigPermission` was intended for dynamic config changes such as `AddCacheConfig`, `AddMapConfig`, ...

WDYT, @vbekiaris ?

Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheCreateConfigMessageTask.java:87]: From the point of view of "what does `CacheCreateConfigMessageTask` do?", the answer is "it creates a `CacheConfig` on all members of the cluster". So it is not any different from `AddCacheConfigMessageTask` that adds `CacheSimpleConfig` (the no-jcache-dependencies Hazelcast configuration equivalent of `CacheConfig`). That's why my first thought was that we should stick to the `ConfigPermission`.

Putting the `CacheCreateConfigMessageTask` in context, it is true that it is invoked as part of `<C extends Configuration> CacheManager.createCache(String, C configuration)` JCache API implementation from a Hazelcast-client-side cache manager. That action creates a client-side `Cache` proxy. Additionally, the `CacheCreateConfigMessageTask` is the only communication with the cluster in the `createCache` implementation - the `Cache` proxy is created locally on the client-side. So if a client does not have the `CREATE` action granted but still is allowed to create a `Cache` proxy on the client-side (because it is granted the `ConfigPermission`) could seem weird behaviour.

bottom line: I am fine with the reasoning for any of `CachePermission(CREATE)` or `ConfigPermission`... maybe the `CachePermission(CREATE)` will lead to less surprising behaviour, so let's keep the change to `CachePermission`.

**Review by vbekiaris — APPROVED** [review]

**Review by vbekiaris — APPROVED** [review]

**Review by yuce — APPROVED** [review]
